packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -346,6 +355,39 @@ class RenderParagraph extends RenderBox
     return _textPainter.computeDistanceToActualBaseline(TextBaseline.alphabetic);
   }
 
+  TextSelection? _textSelection;
+
+  @override
+  bool update(Offset start, Offset end) {
+    final TextPosition startText = getPositionForOffset(start);
+    final TextPosition endText = getPositionForOffset(end);
+    if (startText == endText) {
+      return false;
+    }
+    _textSelection = TextSelection(baseOffset: startText.offset, extentOffset: endText.offset);
+    markNeedsPaint();
+    return true;
```

Comment:
nit: I tried selecting something close to the edge of the bottom of the screen on a web page and the page scrolled to reveal the content below. So maybe call `RenderObject.showOnScreen`.